### PR TITLE
store/tikv: move client*.go into single package tikv/client

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: AndreMouche

### What problem does this PR solve?
This PR moves client*.go into the single package `tikv/client`.

Part of https://github.com/pingcap/tidb/issues/22513

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

File: store/driver/sql_fail_test.go [hunk 1]
```diff
@@ -100,7 +100,6 @@ func (s *testSQLSerialSuite) TestFailBusyServerCop(c *C) {
 }
 
 func TestMain(m *testing.M) {
-	tikv.ReadTimeoutMedium = 2 * time.Second
```
Comment (disksing) [inline store/driver/sql_fail_test.go:103]: why remove this?

File: store/tikv/client/client_test.go [hunk 2]
```diff
@@ -32,6 +33,29 @@ import (
...
+	}
+}
+
+func (s *OneByOneSuite) TearDownSuite(c *C) {
+	if *WithTiKV {
+		withTiKVGlobalLock.Unlock()
+	} else {
+		withTiKVGlobalLock.RUnlock()
+	}
+}
```
Comment (disksing) [inline store/tikv/client/client_test.go:58]: need empty line

File: store/driver/sql_fail_test.go (hunk 1 above)
Comment (AndreMouche) [reply store/driver/sql_fail_test.go:103]: `ReadTimeoutMedium` should be a `const`, but with this line it must be a var. I find it is useless to change this value in this test, so I remove it.

Comment (disksing) [reply store/driver/sql_fail_test.go:103]: Looks it can make tests run longer.

Comment (AndreMouche) [reply store/driver/sql_fail_test.go:103]: I‘ve tried the related test cases on my computer, it runs even faster than before.

**Review by disksing — APPROVED** [review]

**Review by sticnarf — APPROVED** [review]

**Pull request merged**